contracts/src/LiquidityPoolV2.sol

```diff
@@ -29,19 +36,41 @@ contract LiquidityPoolV2 is ReentrancyGuardUpgradeable, AccessManagedUpgradeable
     event LiquidityRemoved(address indexed user, uint256 tokenAAmount, uint256 tokenBAmount);
     event SwapExecuted(address indexed user, address tokenIn, uint256 amountIn, address tokenOut, uint256 amountOut);
 
-    // Initializer function (replaces constructor)
-    function initialize(address _tokenA, address _tokenB, address _initialAuthority) public initializer {
-        require(_tokenA != address(0) && _tokenB != address(0), "LiquidityPool: invalid token addresses");
+    /**
+     * @notice Initializer function (replaces constructor).
+     * @param _tokenA Address of the first token.
+     * @param _tokenB Address of the second token.
+     * @param _accessManager Address of the access manager contract.
+     */
+    function initialize(address _tokenA, address _tokenB, address _accessManager) public initializer {
+        require(_tokenA != address(0), 'LiquidityPool: invalid tokenA address');
+        require(_tokenB != address(0), 'LiquidityPool: invalid tokenB address');
+        require(_accessManager != address(0), 'LiquidityPool: invalid access manager address');
         tokenA = _tokenA;
         tokenB = _tokenB;
-
+        accessManager = IAccessManagerExtended(_accessManager);
         __ReentrancyGuard_init();
-        __AccessManaged_init(_initialAuthority);
+
+        platformFee = 20; // Default platform fee is 0.2%
     }
 
-    // Function to add liquidity to the pool
-    function addLiquidity(uint256 tokenAAmount, uint256 tokenBAmount) external restricted nonReentrant {
-        require(tokenAAmount > 0 && tokenBAmount > 0, "LiquidityPool: amounts must be greater than zero");
+    /**
+     * @notice Modifier to restrict function access to accounts with a specific role.
+     * @param roleId ID of the required role.
+     */
+    modifier onlyRole(uint64 roleId) {
+        (bool hasRole, ) = accessManager.hasRole(roleId, msg.sender);
+        require(hasRole, 'LiquidityPool: sender must have the required role');
+        _;
+    }
+
+    /**
+     * @notice Function to add liquidity to the pool.
+     * @param tokenAAmount Amount of tokenA to add.
+     * @param tokenBAmount Amount of tokenB to add.
+     */
+    function addLiquidity(uint256 tokenAAmount, uint256 tokenBAmount) external onlyRole(1) nonReentrant {
+        require(tokenAAmount > 0 && tokenBAmount > 0, 'LiquidityPool: amounts must be greater than zero');
 
         // Transfer tokens from sender to the contract
         ERC20Upgradeable(tokenA).transferFrom(msg.sender, address(this), tokenAAmount);
```

Comment:
**adjustForDecimals** (see last comment of the file to understand)

Replace tokenAAmount and tokenBAmount in the transfers but the adjustedToken